packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1968,6 +1968,23 @@ class TextInput {
   /// Returns true if a scribble interaction is currently happening.
   bool get scribbleInProgress => _scribbleInProgress;
 
+  void _handleCustomAction(String actionId) {
+    // Support dynamic callbacks (new API)
+    final VoidCallback? callback = SystemContextMenuController._customActionCallbacks[actionId];
+    if (callback != null) {
+      callback();
+      return;
+    }
```

Comment:
Can you also assert that the callback is not null so that developers can be aware that something is wrong?

Reply:
Not sure if this is testable or not, but it would be nice if we can test it.